### PR TITLE
Granitemoe

### PR DESCRIPTION
**Pull request by mayank31398**
This PR adds support for IBM's PowerMoE model (3B)
This PR will also form a basis for IBM's upcoming MoE models by end of this month

text models: @ArthurZucker and @younesbelkada

### CODE REVIEW

**Comment by mayank31398** [comment]
Hi @ArthurZucker any updates on this?

File: src/transformers/models/granitemoe/configuration_granitemoe.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                "`rope_scaling` must be a dictionary with two fields, `type` and `factor`, " f"got {self.rope_scaling}"
+            )
+        rope_scaling_type = self.rope_scaling.get("type", None)
+        rope_scaling_factor = self.rope_scaling.get("factor", None)
+        if rope_scaling_type is None or rope_scaling_type not in ["linear", "dynamic"]:
+            raise ValueError(
+                f"`rope_scaling`'s type field must be one of ['linear', 'dynamic'], got {rope_scaling_type}"
+            )
+        if rope_scaling_factor is None or not isinstance(rope_scaling_factor, float) or rope_scaling_factor <= 1.0:
+            raise ValueError(f"`rope_scaling`'s factor field must be a float > 1, got {rope_scaling_factor}")
```
Comment (ArthurZucker) [inline src/transformers/models/granitemoe/configuration_granitemoe.py:208]:
```suggestion
```

**Review by ArthurZucker — APPROVED** [review]
Let's just add an integration tests with generation!

**Comment by ArthurZucker** [comment]
Thanks, let's merge!

**Comment by mayank31398** [comment]
Thanks

**Pull request merged**